R/Plot_synergy.R

```diff
@@ -1,328 +1,332 @@
+    krig.coord = cbind(seq(1, nc, length = (nc - 1)*(len + 2) - (nc - 2)),
+                       seq(1, nr, length = (nr - 1)*(len + 2) - (nr - 2))),
+    cov.mod = "whitmat",
+    grid = TRUE,
+    sill = 1,
+    range = 10,
+    smooth = 0.8
+  )$krig.est)
+}
+
+.Plot2D <- function(plot.title, mat.tmp, len, row.conc, col.conc,
+                    drug.row, drug.col, start.point, end.point) {
+  # Note a different layout() here than in .PlotAll()
+  layout(matrix(c(1, 2), nrow = 2L, ncol = 1L), heights = c(0.1, 1))
+  .Plot2DTitle(plot.title, start.point, end.point)
+  .Plot2DContourMap(
+    mat.tmp, row.conc, col.conc, drug.row, drug.col, start.point, end.point)
+  return(recordPlot())
+}
+
+.Plot2DTitle <- function (plot.title, start.point, end.point) {
+  par(mar = c(0, 6.1, 2.1, 4.1))
+  suppressWarnings(par(mgp = c(3, -1, 0)))
+  levels <- seq(start.point, end.point, by = 2)
+  plot.new()
+  plot.window(ylim = c(0, 1), xlim = range(levels), xaxs = "i", yaxs = "i")
+  rect(xleft = levels[-length(levels)],
+       ybottom = 0,
+       xright = levels[-1L],
+       ytop = 0.3,
+       col = .GetColors(start.point, end.point),
+       border = NA)
+  title(plot.title)
+  axis(3,
+       tick = FALSE,
+       at = lattice::do.breaks(c(start.point, end.point), end.point/10))
+  par(mar = c(5.1,4.1,2.1,2.1))
+  suppressWarnings(par(mgp = c(2,1,0)))
+}
+
+.Plot2DContourMap <- function(
+  mat.tmp, row.conc, col.conc, drug.row, drug.col, start.point, end.point) {
+  plot.new()
+  x.2D <- (1:dim(mat.tmp)[1] - 1) / (dim(mat.tmp)[1] - 1)
+  y.2D <- (1:dim(mat.tmp)[2] - 1) / (dim(mat.tmp)[2] - 1)
+  plot.window(asp = NA,
+              xlim = range(x.2D),
+              ylim = range(y.2D),
+              log = "",  # no axis uses log scale
+              xaxs = "i",
+              yaxs = "i")
+  # TODO: Find a way to stop depending on private method .filled.contour().
+  .filled.contour(x = x.2D,
+                  y = y.2D,
+                  z = mat.tmp,
+                  levels = seq(start.point, end.point, by = 2),
+                  col = .GetColors(start.point, end.point))
+  box()
+  mtext(drug.col, 1, cex = 1, padj = 3)
+  mtext(drug.row, 2, cex = 1, las = 3, padj = -3)
+
+  axis(side = 1,
+       at = seq(0, 1, by = 1/(length(col.conc) - 1)),
+       labels = round(col.conc, 1))
+  axis(side = 2,
+       at = seq(0, 1, by = 1/(length(row.conc) - 1)),
+       labels = round(row.conc, 1))
+}
+
+# Default values for 'extra' args position and new.page let us take calls via
+# the PlotFunc indirection in .PlotScores(), yet still re-use this function from
+# .PlotAll(), which must specify a layout that places this graph beside the 2D
+# contour plot.
+.Plot3D <- function(
+  plot.title, mat.tmp, len, row.conc, col.conc, drug.row, drug.col,
+  start.point, end.point, position = c(0,0,1,1), new.page = TRUE) {
+  print(
+    lattice::wireframe(
+      x = mat.tmp,
+      scales = list(arrows=FALSE,
```

Comment:
insert space before and after "=", thanks.

Reply:
Done.